math/firep.cpp

```diff
@@ -0,0 +1,570 @@
+/**********************************************************************
+Copyright 2014-2016 The RIVET Developers. See the COPYRIGHT file at
+the top-level directory of this distribution.
+
+This file is part of RIVET.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+**********************************************************************/
+
+#include "firep.h"
+
+#include "index_matrix.h"
+#include "map_matrix.h"
+
+#include "debug.h"
+
+#include <algorithm>
+#include <ctime>
+#include <limits> //std::numeric_limits
+#include <sstream>
+#include <stdexcept>
+
+//FIRep constructor; requires BifiltrationData object (which in particular specifies the homology dimension) and verbosity parameter
+FIRep::FIRep(BifiltrationData& bd, int v)
```

Comment:
This method is much too long for anyone to read comfortably. Please refactor into smaller chunks (add private methods or functions with meaningful names as necessary) so that the end result is a method that fits in ~50 lines or so and reads like a clear explanation of what's going on.